scripts/devaccess/devaccess_wrap.py

```diff
@@ -0,0 +1,454 @@
+            Return remaining string '''
+        cmd_split = cmd.split()
+        delete_tokens = []
+
+        for x in range(0, len(cmd_split)):
+            if cmd_split[x] == '-c':
+                self._params['container'] = cmd_split[x+1]
+
+                # mark tokens for removal
+                delete_tokens.append(cmd_split[x])
+                delete_tokens.append(cmd_split[x+1])
+
+                # skip next token since we already processed it
+                x = x + 1
+            elif cmd_split[x].startswith('-c'):
+                self._params['container'] = re.sub('-c', '', cmd_split[x])
+
+                delete_tokens.append(cmd_split[x])
+
+        for token in delete_tokens:
+            cmd_split.remove(token)
+
+        new_cmd = " ".join(cmd_split)
+        return new_cmd
+
+
+    def get_verb_type_subject(self, cmd):
+        ''' Take command without parameters and parse it into its
+            various components.
+            Return remaining string (should be "")
+        '''
+        cmd_split = cmd.split()
+
+        # handle 'oc <verb> <type> <optional-subject>' type of cmd
+        # ...or even 'oc logs <subject>'
+        cmd_split.remove("oc")
+
+        # make sure none of the remaining tokens are parameters
+        for param in cmd_split:
+            if param.startswith('-'):
+                raise Exception('Should not have any more parameters left for processing')
+
+        self._params['verb'] = cmd_split[0]
+        self._params['type'] = cmd_split[1]
+        if len(cmd_split) > 2:
+            self._params['subject'] = cmd_split[2]
+
+        if self._params['verb'] == 'logs':
+            # logs commands aren't in the form of verb type subject
+            # just verb subject, so wipe out _type and store it in
+            # _subject instead
+            self._params['subject'] = self._params['type']
+            self._params['type'] = None
+
+        cmd_split.remove(self._params['verb'])
+        if self._params['type'] is not None:
+            cmd_split.remove(self._params['type'])
+        if self._params['subject'] is not None:
+            cmd_split.remove(self._params['subject'])
+
+        return " ".join(cmd_split)
+
+    def parse_cmd(self):
+        ''' take a raw oc command and tokenize it '''
+
+        #
+        # Get all params first
+        #
+        cmd_no_namespace = self.get_namespace(self._raw_cmd)
+
+        cmd_no_output_formatting = self.get_output_format(cmd_no_namespace)
+
+        cmd_no_follow = self.get_follow(cmd_no_output_formatting)
+
+        cmd_no_container = self.get_container(cmd_no_follow)
+
+        #
+        # all that is left should be: 'oc <verb> <type> <optional-subject>'
+        #
+        cmd = self.get_verb_type_subject(cmd_no_container)
+
+
+        # should be nothing left after we parsed all the tokens
+        if cmd != "":
+            raise Exception("Unprocessed command tokens left.")
+
+    def normalized_cmd(self, generic=False):
```

Comment:
This function looks to be "oc" specific.  I suggest it has "oc" in the function name to make that clear, and keep namespace tidy for future commands that may need similar processing..